agent/ec2/ec2_metadata_client.go

```diff
@@ -61,100 +58,84 @@ type InstanceIdentityDocument struct {
 	AvailabilityZone string  `json:"availabilityZone"`
 }
 
-type HttpClient interface {
-	Get(string) (*http.Response, error)
-}
-
+// EC2MetadataClient is the EC2 Metadata Service Client used by the
+// ECS Agent
 type EC2MetadataClient interface {
-	DefaultCredentials() (*RoleCredentials, error)
-	ReadResource(string) ([]byte, error)
+	// ReadResource reads the metadata associated with a resource
+	// path from the instance metadata service
+	ReadResource(path string) ([]byte, error)
+	// InstanceIdentityDocument retrieves the instance identity
+	// document from the instance metadata service
 	InstanceIdentityDocument() (*InstanceIdentityDocument, error)
+	// VPCID returns the VPC id for the network interface, given
+	// its mac address
+	VPCID(mac string) (string, error)
+	// SubnetID returns the subnet id for the network interface,
+	// given its mac address
+	SubnetID(mac string) (string, error)
+	// PrimaryENIMAC returns the MAC address for the primary
+	// network interface of the instance
+	PrimaryENIMAC() (string, error)
 }
 
-type ec2MetadataClientImpl struct {
-	client HttpClient
+type ec2MetadataClient struct {
+	httpClient http.Client
 }
 
-func NewEC2MetadataClient(httpClient HttpClient) EC2MetadataClient {
+// NewEC2MetadataClient creates a new ec2MetadataClient object
+func NewEC2MetadataClient(httpClient http.Client) EC2MetadataClient {
 	if httpClient == nil {
-		var lowTimeoutDial http.RoundTripper = &http.Transport{
+		var lowTimeoutDial net_http.RoundTripper = &net_http.Transport{
 			Dial: (&net.Dialer{
-				Timeout: EC2_METADATA_REQUEST_TIMEOUT,
+				Timeout: EC2MetadataRequestTimeout,
 			}).Dial,
 		}
 
-		httpClient = &http.Client{Transport: lowTimeoutDial}
-	}
-
-	return &ec2MetadataClientImpl{client: httpClient}
-}
-
-func (c *ec2MetadataClientImpl) DefaultCredentials() (*RoleCredentials, error) {
-	securityCredentialResp, err := c.ReadResource(SECURITY_CREDENTIALS_RESOURCE)
-	if err != nil {
-		return nil, err
+		httpClient = &net_http.Client{Transport: lowTimeoutDial}
 	}
 
-	securityCredentialList := strings.Split(strings.TrimSpace(string(securityCredentialResp)), "\n")
-	if len(securityCredentialList) == 0 {
-		return nil, errors.New("No security credentials in response")
-	}
-
-	defaultCredentialName := securityCredentialList[0]
-
-	rawResp, err := c.ReadResource(SECURITY_CREDENTIALS_RESOURCE + defaultCredentialName)
-	if err != nil {
-		return nil, err
-	}
-	var credential RoleCredentials
-	err = json.Unmarshal(rawResp, &credential)
-	if err != nil {
-		return nil, err
-	}
-	return &credential, nil
+	return &ec2MetadataClient{httpClient: httpClient}
 }
 
-func (c *ec2MetadataClientImpl) InstanceIdentityDocument() (*InstanceIdentityDocument, error) {
-	rawIidResp, err := c.ReadResource(INSTANCE_IDENTITY_DOCUMENT_RESOURCE)
+func (c *ec2MetadataClient) InstanceIdentityDocument() (*InstanceIdentityDocument, error) {
```

Comment:
That would be better. But, this is legacy code. I think the better thing to do is to get rid of most of this file itself. I think this is outside the scope of this PR as well.